test/parallel/test-stream-writable-destroy.js

```diff
@@ -344,3 +344,44 @@ const assert = require('assert');
   }));
   write.destroy(new Error('asd'));
 }
+
+{
+  // Call buffered write callback with error
+
+  const write = new Writable({
+    write(chunk, enc, cb) {
+      process.nextTick(cb, new Error('asd'));
+    },
+    autoDestroy: false
+  });
+  write.cork();
+  write.write('asd', common.mustCall((err) => {
+    assert.strictEqual(err.message, 'asd');
+  }));
+  write.write('asd', common.mustCall((err) => {
+    assert.strictEqual(err.code, 'ERR_STREAM_DESTROYED');
+  }));
+  write.on('error', common.mustCall((err) => {
+    assert.strictEqual(err.message, 'asd');
+  }));
+  write.uncork();
+}
+
+{
+  // Ensure callback order.
+
+  let state = 0;
+  const write = new Writable({
+    write(chunk, enc, cb) {
+      process.nextTick(cb);
```

Comment:
```suggestion
      // `setImmediate` is used on purpose to ensure the callback is called
      // after `process.nextTick()` callbacks. 
      setImmediate(cb);
```

Otherwise it does not fail on current master.

Reply:
Thanks!